modules/auxiliary/scanner/rservices/rexec_login.rb

```diff
@@ -179,9 +179,11 @@ def start_rexec_session(host, port, user, pass, proof, stderr_sock)
       :stderr_sock    => stderr_sock
     }
 
-    # Don't tie the life of this socket to the exploit
-    self.sockets.delete(stderr_sock)
-
-    start_session(self, "rexec #{user}:#{pass} (#{host}:#{port})", merge_me) if datastore['CreateSession']
+    if datastore['CreateSession']
+      start_session(self, "rexec #{user}:#{pass} (#{host}:#{port})", merge_me, false, self.sock)
```

Comment:
This change was required by the changes to https://github.com/rapid7/metasploit-framework/pull/15831


`obj.respond_to?(:sock)` is false, as it's not a publicly accessible field. However `self` in the `start_session` context is the same as `obj`, so `sock` was previously accessible

```diff
-    sock ||= obj.sock
+     sock ||= obj.respond_to?(:sock) ? obj.sock : nil
     sess ||= Msf::Sessions::CommandShell.new(sock)
     sess.set_from_exploit(obj)
    sess.info = info
```

 https://github.com/rapid7/metasploit-framework/pull/15831/files#diff-094966c8b83cae1221563db0129e812247847cf22d73d0b92e8acf4a91e607ef changed